### PR TITLE
LPS-137162 Add tests

### PR DESCRIPTION
**Pull request by SamZiemer**
Hi @achaparro,

Here are some starting tests for the report functionality so far. A couple of notes.

https://github.com/achaparro/liferay-portal/compare/LPS-137162...SamZiemer:LPS-137162?expand=1#diff-a34012faa33af23a7e9613254fc16bc6ca3111e17fd388baa3d068e5afc64718 This file only contains a snippet of what we generate in the report as it is the only information we can guarantee to be accurate across testing environments. Though I could add more information regarding the properties we check in the tests, I don't think that would prove any better functionality than what this already does.

Secondly, Ideally we would remove the report file created in the test along with the directory, but I did not find a good way to do that without it being clunky. One option I thought could work is simple delete them in the [last test method](https://github.com/achaparro/liferay-portal/compare/LPS-137162...SamZiemer:LPS-137162?expand=1#diff-aea7070d0ff0ab733907a509bbfff6e4b23042f7773b4035f65037f6ff31109bR106), but I was not sure if that was the right thing to do.

Lastly, the set up and tear down methods, since those run before and after every test, so the tests generate a lot of files and adjusting the release record for every test, is there a better way to approach this?

### CODE REVIEW

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:reopen

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
Regarding the source formatter test failure, when I run it on my local branch, the SF passes with no changes. I'm looking into why it fails when testing with CI.

**Comment by SamZiemer** [comment]
ci:test:relevant

File: modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/test/UpgradeReportLogAppenderTest.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		_initialSchemaVersion = release.getSchemaVersion();
+
+		release = _releaseLocalService.getRelease(1);
+
+		release.setBuildNumber(ReleaseInfo.RELEASE_7_1_0_BUILD_NUMBER);
+		release.setSchemaVersion("1.0.0");
+
+		_releaseLocalService.updateRelease(release);
+
+		_appender.start();
```
Comment (achaparro) [inline modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/test/UpgradeReportLogAppenderTest.java:67]: The report should be executed in every test so we can execute different use cases per test. You can capture in the setUp method the initial values of the schemaVersion and properties

File: modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/test/UpgradeReportLogAppenderTest.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		Release release = _releaseLocalService.getRelease(1);
+
+		release.setBuildNumber(_initialBuildNumber);
+		release.setSchemaVersion(_initialSchemaVersion);
+
+		_releaseLocalService.updateRelease(release);
```
Comment (achaparro) [inline modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/test/UpgradeReportLogAppenderTest.java:86]: The tearDown is good, just restore any value you may change in the tests. You may also have to remove the report folder here.

File: modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/test/UpgradeReportLogAppenderTest.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	}
+
+	@Test
+	public void testReportContents() throws Exception {
+		File reportsDir = new File(".", "reports");
+
+		File reportFile = new File(reportsDir, "upgrade_report.info");
+
+		Assert.assertThat(
+			FileUtil.read(reportFile),
```
Comment (achaparro) [inline modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/test/UpgradeReportLogAppenderTest.java:112]: Since the properties can change in time, you have two options here:
- or modify them in the setUp and put them in the upgrade_report template 
- or take the value here and add modify the content for the template adding the lines for the properties

**Review by achaparro — COMMENTED** [review]
Hi @SamZiemer,

The tests look promising however, they are not working (see failures in the CI) because we always print the properties and that's not present in the template you added. Are you still having issues testing locally?

Regarding your questions:
> Lastly, the set up and tear down methods, since those run before and after every test, so the tests generate a lot of files and adjusting the release record for every test, is there a better way to approach this?

You can use setUpClass for anything that needs to be changed once by class. The same happens for tearDownClass

> Secondly, Ideally we would remove the report file created in the test along with the directory, but I did not find a good way to do that without it being clunky. One option I thought could work is simple delete them in the last test method, but I was not sure if that was the right thing to do.

Can't you remove the report folder in the tearDown method (we should generate the report in every test method)

File: modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/test/UpgradeReportLogAppenderTest.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+		Assert.assertTrue(reportsDir.exists());
+	}
+
+	@Test
+	public void testGenerateReportFile() {
+		File reportsDir = new File(".", "reports");
+
+		File reportFile = new File(reportsDir, "upgrade_report.info");
+
+		Assert.assertTrue(reportFile.exists());
```
Comment (achaparro) [inline modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/test/UpgradeReportLogAppenderTest.java:102]: This two test methods are ok but not really important because you are testing the existence of the folder and file in the  following method testReportContents. You can add more use cases for the reportContents, like for example:
- testReportContentWithDefaultValues when executing the test without change anything
- testReportContent with the changes you are doing.

It is up to you but try to cover both most common cases and edge cases.
